capa/features/extractors/vmray/__init__.py

```diff
@@ -185,7 +189,17 @@ def _compute_monitor_processes(self):
                 self.monitor_processes[monitor_process.process_id] = vmray_monitor_process
             else:
                 # we expect monitor processes recorded in both SummaryV2.json and flog.xml to equal
-                assert self.monitor_processes[monitor_process.process_id] == vmray_monitor_process
+                # to ensure this, we compare the pid, monitor_id, and origin_monitor_id
+                # for the other fields we've observed cases with slight deviations, e.g.,
+                # the ppid for a process in flog.xml is not set correctly, all other data is equal
+                sv2p = self.monitor_processes[monitor_process.process_id]
+                assert (sv2p.pid, sv2p.monitor_id, sv2p.origin_monitor_id) == (
+                    vmray_monitor_process.pid,
+                    vmray_monitor_process.monitor_id,
+                    vmray_monitor_process.origin_monitor_id,
+                )
```

Comment:
thoughts on being even more lenient here and not asserting this but just reporting it?
I'm encountering more inconsistencies between the two files, e.g., the `monitor_id` not being set

Reply:
so we could do the check and just log it

Reply:
I'm hesitant to be lenient here because we rely on sane process and thread monitor IDs for indexing. I'd consider differences between _SummaryV2.json_ and _flog.xml_ to be VMRay bugs and, if true, can we trust capa's results at that point?

Reply:
Right, let's leave as is until we find more samples that fail.